opentelemetry/proto/profiles/v1/profiles.proto

```diff
@@ -182,6 +182,9 @@ message AttributeSet {
 // A stacktrace is a sequence of locations. Order of locations goes from callers to callees. Many stacktraces will point to the same locations. The link between stacktraces, attribute sets, links, values and timestamps is implicit and is based on the order of the elements in the corresponding tables in ProfileType message.
 message Stacktrace {
   repeated uint32 location_indices = 1;
+
+  // An 128bit id that uniquely identifies this stacktrace, globally. Index into string table.
+  uint32 trace_id_index;
```

Comment:
Introducing a well-specified and globally unique `Stacktrace` id may help interoperability. The `Stacktrace` id can be calculated by hashing, for every `Location` in the `Stacktrace`:

1. `The file_id` corresponding to `Location.Mapping`
2. The address of the `Location` translated into a stable executable offset:  `Location.Mapping.file_offset + (Location.address - Location.Mapping.memory_start)`

An id for mixed `Stacktraces` can also be generated by special-casing `Locations` containing non-"native" types (see: `type_index` below):

1. Instead of the `file_id` of `Location.Mapping`, a "synthetic" `file_id` can be generated by using a runtime-specific scheme (e.g. hash of the filename)
2. Instead of `Location.address`, we can use line numbers.